mmdet/apis/inference.py

```diff
@@ -65,16 +65,28 @@ def init_detector(
         elif 'CLASSES' in checkpoint_meta:
             # < mmdet 3.x
             classes = checkpoint_meta['CLASSES']
-            model.dataset_meta = {'CLASSES': classes, 'PALETTE': palette}
+            model.dataset_meta = {'CLASSES': classes}
         else:
             warnings.simplefilter('once')
             warnings.warn(
                 'dataset_meta or class names are not saved in the '
                 'checkpoint\'s meta data, use COCO classes by default.')
-            model.dataset_meta = {
-                'CLASSES': get_classes('coco'),
-                'PALETTE': palette
-            }
+            model.dataset_meta = {'CLASSES': get_classes('coco')}
+
+    # Priority:  args.palette -> config -> checkpoint
+    if palette != 'none':
+        model.dataset_meta['PALETTE'] = palette
+    else:
+        metainfo = config.test_dataloader.dataset.get('metainfo', {})
+        cfg_palette = metainfo.get('PALETTE', None)
+        if cfg_palette is not None:
+            model.dataset_meta['PALETTE'] = cfg_palette
+        else:
+            if 'PALETTE' not in model.dataset_meta:
+                warnings.warn(
+                    'PALETTE does not exist, coco is used by default. '
+                    'You can also set the palette to customize.')
+                model.dataset_meta['PALETTE'] = 'coco'
```

Comment:
If there is no 'PALETTE', should the color be randomly set? If set it to coco by default, will it cause some error if the length of PALETTE smaller than the num_classes of the model?

Reply:
That's right, but this one is kept consistent with the previous implementation.